src/LayoutComponents/nav/collapsible.scss

```diff
@@ -0,0 +1,52 @@
+@mixin menu-transition {
+  transition: transform 250ms ease-in-out, height 250ms ease-in-out, opacity 250ms ease-in-out ;
+}
+
+@mixin open {
+  height: 100%;
+  opacity: 1;
+  transform: translate(0px, 0%) scaleY(1);
+}
+
+@mixin closed {
+  height: 0px;
+  opacity: 0;
+  transform: translate(0px, -45%) scaleY(0.1);
+}
+
+.collapse-button {
+  background-color: transparent;
+  border: none;
+  color: #006400;
+  font-size: 45px;
+}
+
+.collapse-menu {
+  @include menu-transition;
+
+  &.entering {
+    @include closed;
+  }
+  &.entered {
+    @include open;
+  }
+  &.exiting {
+    @include open;
+  }
+  &.exited {
+    @include closed;
+  }
+}
```

Comment:
This can be shortened:
```css
.collapse-menu {
  transition: all 250ms ease-in-out;
  transition-property: transform, height, opacity;

  &.entering, &.exited {
    height: 0px;
    opacity: 0;
    transform: translate(0px, -45%) scaleY(0.1);
  }
  &.entered, &.exiting {
    height: 100%;
    opacity: 1;
    transform: translate(0px, 0%) scaleY(1);
  }
}
```
The mixins would not be needed anymore

Reply:
To add to that, there are a couple more things we can cut out here. You might have noticed that you can't transition to a relative height. This element usually has a fixed height, however. Switching to the following makes it so that the animation is smooth, the content doesn't jump and the menu items' shape doesn't distort:
```css
.collapse-menu {
  transition: height 250ms ease-in-out;
  overflow: hidden;

  &.entering, &.exited {
    height: 0;
  }
  &.entered, &.exiting {
    height: 95vh;
  }
}
```